packages/@react-aria/toast/src/useToastRegion.ts

```diff
@@ -29,14 +29,51 @@ export function useToastRegion<T>(props: AriaToastRegionProps, state: ToastState
   let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-aria/toast');
   let {landmarkProps} = useLandmark({
     role: 'region',
-    'aria-label': props['aria-label'] || stringFormatter.format('notifications')
+    'aria-label': props['aria-label'] || stringFormatter.format('notifications', {count: state.visibleToasts.length})
   }, ref);
 
   let {hoverProps} = useHover({
     onHoverStart: state.pauseAll,
     onHoverEnd: state.resumeAll
   });
 
+  // Manage focus within the toast region.
+  // If a focused containing toast is removed, move focus to the next toast, or the previous toast if there is no next toast.
+  // We might be making an assumption with how this works if someone implements the priority queue differently, or
+  // if they only show one toast at a time.
+  let toasts = useRef([]);
+  let prevVisibleToasts = useRef(state.visibleToasts);
+  let focusedToast = useRef(null);
+  useLayoutEffect(() => {
+    // If no toast has focus, then don't do anything.
+    if (focusedToast.current === -1) {
+      return;
+    }
+    toasts.current = [...ref.current.querySelectorAll('[role="alertdialog"]')];
+    // If the visible toasts haven't changed, we don't need to do anything.
+    if (prevVisibleToasts.current.length === state.visibleToasts.length
+      && state.visibleToasts.every((t, i) => t.key === prevVisibleToasts.current[i].key)) {
+      return;
+    }
+    // Get a list of all removed toasts by index.
+    let removedToasts = prevVisibleToasts.current
+      .map((t, i) => ({...t, i}))
+      .filter(((t) => !state.visibleToasts.some(t2 => t.key === t2.key)));
+
+    // If the focused toast was removed, focus the next or previous toast.
+    if (removedToasts.some(t => t.i === focusedToast.current)) {
+      let nextToast = focusedToast.current;
+      if (nextToast >= toasts.current.length) {
+        nextToast = toasts.current.length - 1;
+      }
+      if (nextToast !== -1) {
+        focusWithoutScrolling(toasts.current[nextToast]);
```

Comment:
We may want to wait a frame before restoring focus here, so that the list item count does not still include the item being removed.